index.ts

```diff
@@ -0,0 +1,72 @@
+import { spawn, SpawnOptionsWithoutStdio } from "child_process";
+import type { Plugin as VitePlugin } from "vite";
+
+// Utility to invoke a given sbt task and fetch its output
+function printSbtTask(task: string, cwd?: string): Promise<string> {
+  const args = ["--batch", "-no-colors", "-Dsbt.supershell=false", `print ${task}`];
+  const options: SpawnOptionsWithoutStdio = {
+    cwd: cwd,
+  };
+  const child = process.platform === 'win32'
+    ? spawn("sbt.bat", args.map(x => `"${x}"`), {shell: true, ...options})
```

Comment:
Have you considered the following options for win:

> windowsVerbatimArguments [<boolean>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type) No quoting or escaping of arguments is done on Windows. Ignored on Unix. This is **set to true automatically when shell is specified** and is CMD. Default: false.
>
> windowsHide [<boolean>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type) Hide the subprocess console window that would normally be created on Windows systems. Default: false.

Reply:
I tried many different things on Windows.

`windowsVerbatimArguments` did not seem to make a difference.

I did not try `windowsHide` but the default behavior seems to do what I want, because we're already running in a console, so it executes in the same console window.